compiler/sem/sigmatch.nim

```diff
@@ -2919,13 +2924,13 @@ proc matchesAux(c: PContext, n, nOrig: PNode, m: var TCandidate, marker: var Int
 
         case operand.kind
         of nkError:
-          arg = operand
+          discard "assigned below"
```

Comment:
It's a pre-existing issues (so no change required), but since the typed storage (`n`) is not updated with the error, the remaining overloads expecting typed parameters, will re-sem the AST, resulting in potentially different errors.

Reply:
Yup, good analysis. It definitely seemed fishy that I wasn't assigning the analysed result, but I didn't get a chance to think through all the consequences. I'll fix that in a separate PR, which combined with this change should slim down `matchesAux` a bit.

Reply:
Follow-on work done in this PR: https://github.com/nim-works/nimskull/pull/1133